### PR TITLE
Batch size 8 (fewer batches per epoch, potentially more epochs)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
With batch_size=4, each epoch has ~203 batches at ~31s/epoch. Doubling to batch_size=8 halves the batches per epoch (~102), which should make each epoch ~50% faster. This could yield 15+ epochs in 5 minutes instead of 10. More epochs means more training, even though each step sees fewer gradient updates. The cosine schedule (T_max=10) would need adjusting to T_max=15 to match.

**Risk:** Fewer gradient updates per epoch, noisier initial training. VRAM should be fine (15.4→~30GB on 96GB GPU).

### 1. Change batch_size (line 29):
```python
# OLD:
    batch_size: int = 4

# NEW:
    batch_size: int = 8
```

### 2. Adjust T_max for more epochs. Replace line 83:
```python
# OLD:
scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max=10)

# NEW:
scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max=15)
```

### 3. Adjust curriculum transition for 15 epochs (line 104):
```python
# OLD:
    alpha = min(epoch / 4.0, 1.0)  # alpha=1.0 at epoch 4

# NEW:
    alpha = min(epoch / 6.0, 1.0)  # alpha=1.0 at epoch 6 (~40% of 15 epochs)
```

Run:
```bash
python train.py --batch_size 8 --agent fern --wandb_group mar14b-bs8 --wandb_name "fern/bs8-tmax15"
```

## Baseline
| Metric | Value |
|--------|-------|
| **surf_p** | **107.35** |
| surf_Ux | 1.23 |
| surf_Uy | 0.84 |
| epochs | ~10 (~31s/ep) |

---

## Results

**W&B run**: `ol0dcqjk`
**Epochs completed**: 11 (wall-clock ~5.3 min)
**Best epoch**: 11
**Peak VRAM**: ~15.4 GB
**Epoch time**: 28.2s (vs 31s baseline, only **-9%** — not the expected ~50%)

| Metric | bs=8 | Baseline (bs=4) | Delta |
|---|---|---|---|
| val/loss | 2.5654 | 2.45 | +0.12 |
| Surface MAE Ux | 1.22 | 1.23 | -0.01 |
| Surface MAE Uy | 0.86 | 0.84 | +0.02 |
| Surface MAE p | 116.42 | **107.35** | **+9.07 (+8.5%)** |
| Volume MAE Ux | 5.64 | 5.44 | +0.20 |
| Volume MAE Uy | 2.57 | 2.57 | 0 |
| Volume MAE p | **149.2** | 152.5 | -3.3 |
| Epoch time | 28.2s | 31.0s | -9% |
| Epochs in 5min | 11 | 10 | +1 |

**What happened**: The speedup hypothesis did not materialize — epoch time dropped only 9% (28.2s vs 31s), not the expected 50%. This suggests the bottleneck is not batch size but data loading or model compute per sample. Only 11 epochs were achieved vs the hoped 15. More importantly, surf_p regressed significantly (107.35 → 116.42, +8.5%). The fewer gradient updates per epoch (1122 steps vs 2030) appear to hurt learning quality. The additional epoch gained does not compensate for the per-step quality loss.

**Suggested follow-ups**:
- batch_size=4 is confirmed better. The model benefits from more gradient steps (2030 vs 1122 per run).
- The epoch time bottleneck is likely in the attention computation or data loading, not batch processing overhead.

### CODE REVIEW

**Comment by tcapelle** [comment]
Closing — bs=8 gave only 11 epochs (vs 10 at bs=4). surf_p=116.4, worse than baseline. Larger batch hurts learning efficiency.
